### PR TITLE
Add ability to set asset init options

### LINKED ISSUE #68: Add option dictionary to the AVURLAsset.
**Is your feature request related to a problem? Please describe.**
Add a way to supply option key-values to the AVURLAsset by using [init(url:options:)](https://developer.apple.com/documentation/avfoundation/avurlasset/1385698-init).

**Describe the solution or feature you would like**
Should be able to supply these values to the `AudioPlayer`, which will pass it down to the `AVPlayerWrapper` that initializes the `AVURLAsset`.

**Describe alternatives you've considered**
A protocol that `AudioItem`s can conform to in order to supply relevant options.

**Additional context**
A result of PR #64, that was adding auth headers to the asset by supplying headers in the option dictionary with the `AVURLAssetHTTPHeaderFieldsKey` key, but this is a private API. The feature described here offers a way for developers to use this at their own risk.

### PR DESCRIPTION
**Pull request by dcvz**
Closes #68

### CODE REVIEW

File: SwiftAudio/Classes/AudioItem.swift [hunk 1]
```diff
@@ -36,6 +36,11 @@ public protocol InitialTiming {
     func getInitialTime() -> TimeInterval
 }
 
+/// Make your `AudioItem`-subclass conform to this protocol to control enable the ability to set initialization options for the asset.
```
Comment (jorgenhenrichsen) [inline SwiftAudio/Classes/AudioItem.swift]: Make your `AudioItem`-subclass conform to this protocol ~~to control enable the ability~~ to set initialization options for the asset.

File: SwiftAudio/Classes/AudioItem.swift [hunk 2]
```diff
@@ -125,3 +130,24 @@ public class DefaultAudioItemInitialTime: DefaultAudioItem, InitialTiming {
     }
     
 }
+
+/// An AudioItem that also conforms to the `AssetOptionsProviding`-protocol
+public class DefaultAudioItemAuthorizing: DefaultAudioItem, AssetOptionsProviding {
```
Comment (jorgenhenrichsen) [inline SwiftAudio/Classes/AudioItem.swift]: Change the naming of the class to `DefaultAudioItemAssetOptionsProviding`.

File: SwiftAudio/Classes/AudioItem.swift [hunk 3]
```diff
@@ -36,6 +36,11 @@ public protocol InitialTiming {
     func getInitialTime() -> TimeInterval
 }
 
+/// Make your `AudioItem`-subclass conform to this protocol to control enable the ability to set initialization options for the asset.
+public protocol AssetOptionsProviding {
+    func getAssetOptions() -> [String : Any]
```
Comment (jorgenhenrichsen) [inline SwiftAudio/Classes/AudioItem.swift]: [String: Any] for consistency.

File: SwiftAudio/Classes/AudioItem.swift [hunk 4]
```diff
@@ -125,3 +130,24 @@ public class DefaultAudioItemInitialTime: DefaultAudioItem, InitialTiming {
...
+        self.options = [:]
+        super.init(audioUrl: audioUrl, artist: artist, title: title, albumTitle: albumTitle, sourceType: sourceType, artwork: artwork)
+    }
+    
+    public init(audioUrl: String, artist: String?, title: String?, albumTitle: String?, sourceType: SourceType, artwork: UIImage?, options: [String: Any]) {
+        self.options = options
+        super.init(audioUrl: audioUrl, artist: artist, title: title, albumTitle: albumTitle, sourceType: sourceType, artwork: artwork)
+    }
+    
+    public func getAssetOptions() -> [String : Any] {
```
Comment (jorgenhenrichsen) [inline SwiftAudio/Classes/AudioItem.swift]: [String: Any] for consistency.

File: SwiftAudio/Classes/AVPlayerWrapper/AVPlayerWrapper.swift [hunk 5]
```diff
@@ -19,7 +19,6 @@ public enum PlaybackEndedReason: String {
 }
 
 class AVPlayerWrapper: AVPlayerWrapperProtocol {
-
```
Comment (jorgenhenrichsen) [inline SwiftAudio/Classes/AVPlayerWrapper/AVPlayerWrapper.swift]: Probably not meant to remove this newline.

File: SwiftAudio/Classes/AudioItem.swift (hunk 1 above)
Comment (jorgenhenrichsen) [reply SwiftAudio/Classes/AudioItem.swift]: Maybe also link to these [docs](https://developer.apple.com/documentation/avfoundation/avurlasset/initialization_options) to make it clear what options are to be provided.

**Comment by dcvz** [comment]
Fixed the issues @jorgenhenrichsen :) obviously I made this PR a bit too fast. Should all be good now!

**Pull request merged**